metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2216,6 +2216,16 @@ public void addDynamicPartitions(long txnId, String dbName, String tableName,
   }
 
   @Override
+  public void preInsertTable(Table table, boolean overwrite) throws MetaException {
```

Comment:
Better to pass in PreInsertTableDesc object instead of 2 arguments. Helps in evolving interface better.